src/frontend/src/lib/api/signer.api.ts

```diff
@@ -1,22 +1,42 @@
 import type { BitcoinNetwork, SignRequest } from '$declarations/signer/signer.did';
-import { getSignerActor } from '$lib/actors/actors.ic';
-import type { EthAddress } from '$lib/types/address';
+import { SignerCanister } from '$lib/canisters/signer.canister';
+import { SIGNER_CANISTER_ID } from '$lib/constants/app.constants';
+import type { BtcAddress, EthAddress } from '$lib/types/address';
 import type { OptionIdentity } from '$lib/types/identity';
+import { Principal } from '@dfinity/principal';
+import { assertNonNullish, isNullish } from '@dfinity/utils';
+
+let canister: SignerCanister | undefined = undefined;
 
 export const getBtcAddress = async ({
 	identity,
 	network
 }: {
 	identity: OptionIdentity;
 	network: BitcoinNetwork;
-}): Promise<EthAddress> => {
-	const { caller_btc_address } = await getSignerActor({ identity });
-	return caller_btc_address(network);
+}): Promise<BtcAddress> => {
+	const { getBtcAddress } = await signerCanister({ identity });
+
+	return getBtcAddress({ network });
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const updateBtcBalance = async ({
```

Comment:
can you provide this in another PR. Let's stay focus on the canister part in this PR.

Plus it will need to use another instance of the canister if you want to use the call in a web worker.